core/sql/optimizer/RelSample.cpp

```diff
@@ -925,7 +925,6 @@ RelExpr *RelSample::bindNode(BindWA *bindWA)
   RelExpr* myChild = child(0);
   if (myChild->getOperatorType() == REL_SCAN &&
       (static_cast<Scan*>(myChild))->isHbaseTable() &&
-      myChild->selectionPred().entries() == 0 &&
```

Comment:
It doesn't really matter where the predicate is applied.  Sampling is just like another predicate ANDed to any existing conjuncts. Whether you sample first or last, the estimated cardinality is the same, and all constraints imposed by the predicate will be satisfied by the selected rows.